src/codegen/generate.ts

```diff
@@ -21,6 +21,34 @@ type ContentType = "json" | "form" | "multipart";
 type OnlyMode = "readOnly" | "writeOnly";
 type OnlyModes = Record<OnlyMode, boolean>;
 
+// Use union of OAS 3.0 and 3.1 types throughout
+type OpenAPISchemaObject = OpenAPIV3.SchemaObject | OpenAPIV3_1.SchemaObject;
+type OpenAPIReferenceObject =
+  | OpenAPIV3.ReferenceObject
+  | OpenAPIV3_1.ReferenceObject;
+type OpenAPIParameterObject =
+  | OpenAPIV3.ParameterObject
+  | OpenAPIV3_1.ParameterObject;
+type OpenAPIDocument = OpenAPIV3.Document | OpenAPIV3_1.Document;
+type OpenAPIDiscriminatorObject =
+  | OpenAPIV3.DiscriminatorObject
+  | OpenAPIV3_1.DiscriminatorObject;
+type OpenAPIResponseObject =
+  | OpenAPIV3.ResponseObject
+  | OpenAPIV3_1.ResponseObject;
+type OpenAPIResponsesObject =
+  | OpenAPIV3.ResponsesObject
+  | OpenAPIV3_1.ResponsesObject;
+type OpenAPIRequestBodyObject =
+  | OpenAPIV3.RequestBodyObject
+  | OpenAPIV3_1.RequestBodyObject;
+type OpenAPIMediaTypeObject =
+  | OpenAPIV3.MediaTypeObject
+  | OpenAPIV3_1.MediaTypeObject;
+type OpenAPIOperationObject =
+  | OpenAPIV3.OperationObject
+  | OpenAPIV3_1.OperationObject;
```

Comment:
This could well be extracted into a separate file for my taste, but all other types seem to be in here as well, so I left it for now.